op-node/rollup/attributes/engine_consolidate.go

```diff
@@ -82,31 +82,41 @@ func checkParentBeaconBlockRootMatch(attrRoot, blockRoot *common.Hash) error {
 	return nil
 }
 
-func checkWithdrawalsMatch(attrWithdrawals *types.Withdrawals, blockWithdrawals *types.Withdrawals) error {
-	if attrWithdrawals == nil && blockWithdrawals == nil {
-		return nil
+// checkWithdrawals checks if the withdrawals list and withdrawalsRoot are as expected in the attributes and block,
+// based on the active hard fork.
```

Comment:
This logic is difficult to read and ensure that it is correct

Reply:
I agree. Had to re-read this function several times and I'm not 100% sure it's correct. Consider creating a separate function to check withdrawals for an isthmus. Some code will get duplicated but it'll be much clearer what the rules are.